src/supertux/menu/addon_browse_menu.cpp

```diff
@@ -0,0 +1,240 @@
+//  SuperTux
+//  Copyright (C) 2022 Vankata453
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "supertux/menu/addon_browse_menu.hpp"
+
+#include <fmt/format.h>
+
+#include "addon/addon.hpp"
+#include "addon/addon_manager.hpp"
+#include "gui/dialog.hpp"
+#include "gui/menu_item.hpp"
+#include "gui/menu_manager.hpp"
+#include "supertux/menu/addon_preview_menu.hpp"
+#include "supertux/menu/download_dialog.hpp"
+#include "util/log.hpp"
+
+namespace {
+
+#define IS_REPOSITORY_MENU_ID(idx) (((idx) - MNID_ADDON_LIST_START) % 2 == 0)
+
+#define MAKE_REPOSITORY_MENU_ID(idx) (MNID_ADDON_LIST_START + 2 * (idx) + 0)
+
+#define UNPACK_REPOSITORY_MENU_ID(idx) ((((idx) - MNID_ADDON_LIST_START) - 0) / 2)
+
+} // namespace
+
+AddonBrowseMenu::AddonBrowseMenu(const bool langpacks_only, const bool auto_install_langpack) :
+  m_addon_manager(*AddonManager::current()),
```

Comment:
`const` not necessary for primitives, only for objects (which are passed by reference).